test/functional/ex_cmds/grep_spec.lua

```diff
@@ -18,4 +18,20 @@ describe(':grep', function()
     feed('<cr>')  -- Press ENTER
     ok(eval('len(getqflist())') > 9000)  -- IT'S OVER 9000!!1
   end)
+
+  it('works when both shellpipe and makeef are unset', function()
+    if eval("executable('grep')") == 0 then
+      pending('missing "grep" command')
+      return
+    end
+
+    feed_command([[set grepprg=grep\ -r]])
+    feed_command([[set shellpipe=]])
+    feed_command([[set makeef=]])
+    -- Change to test directory so that the test does not run too long.
+    feed_command('cd test')
+    feed_command('grep a **/*')
+    feed('<cr>')  -- Press ENTER
+    ok(eval('len(getqflist())') > 9000)  -- IT'S OVER 9000!!1
```

Comment:
this test is not for UI performance so we could use a more restricted `grep` pattern above. So this test will take less time, and we can assert something more explicit.

Reply:
The description of the test includes "large output" though. Can you tell what the minimum thing is we should test? Regardless, I'd prefer fixing that up in a separate commit.